internal/state/graph/httproute.go

```diff
@@ -623,3 +639,67 @@ func validateFilter(
 
 	return allErrs
 }
+
+func validateFilterHeaderModifier(
+	validator validation.HTTPFieldsValidator,
+	filter v1beta1.HTTPRouteFilter,
+	filterPath *field.Path,
+) field.ErrorList {
+	headerModifier := filter.RequestHeaderModifier
+
+	headerModifierPath := filterPath.Child("requestHeaderModifier")
+
+	if headerModifier == nil {
+		panicForBrokenWebhookAssumption(errors.New("requestHeaderModifier cannot be nil"))
+	}
+
+	if len(headerModifier.Add) == 0 && len(headerModifier.Set) == 0 && len(headerModifier.Remove) == 0 {
+		panicForBrokenWebhookAssumption(errors.New("requestHeaderModifier requires at least one of Add, Set, or Remove"))
+	}
+
+	return validateFilterHeaderModifierFields(validator, headerModifier, headerModifierPath)
+}
+
+func validateFilterHeaderModifierFields(
+	validator validation.HTTPFieldsValidator,
+	headerModifier *v1beta1.HTTPHeaderFilter,
+	headerModifierPath *field.Path,
+) field.ErrorList {
+	var allErrs field.ErrorList
+	// FIXME: @ciarams87 Ensure duplicate header names are not listed more than once
+
+	if len(headerModifier.Add) > 0 {
```

Comment:
Same comment as before, length check isn't needed if we're ranging over the list.